File: java/03_library_native_jni/source/src/main/java/namespace/Comparator.java
${{VAR_COPYRIGHT_HEADER}}

${{VAR_NAMESPACE_PACKAGE_DECLARATION}}

/**
 * Dummy interface.
 *
 * @param <T> The type of objects to compare.
 *
 */
public interface Comparator<T> {

    /**
     * Compares the specified object to the set object of this Comparator.
     *
     * @param val The object to compare.
     *
     * @return An int indicating the result of the comparison.
     *         Returns 0 (zero) if both objects are equal.
     *         Returns a negative int value if the set object is less
     *         than the specified object. Returns a positive int value if
     *         the set object is greater than the specified object.
     */
    public int compare(T val);

}

File: python/03_library_native_ctypes/source/package/comparator.py
${{VAR_COPYRIGHT_HEADER}}
"""Module providing a Comparator class."""

from abc import ABC, abstractmethod


class Comparator(ABC):
    """Dummy abstract base class.

    Attributes:
        val (str): The base value used by the Comparator.
        ncalls (int): The number of calls to the `compare()` method of
            the Comparator instance.
    """

    def __init__(self, val: str):
        self.val: str = val
        self.ncalls: int = 0

    @abstractmethod
    def compare(self, val: str) -> int:
        """Compares the specified object to the set object of this Comparator.

        Args:
            val (str): The string with which to compare.

        Returns:
            int: An int indicating the result of the comparison.
                0 (zero) if both objects are equal.
                A negative int value if the set object is less than the
                specified object.
                A positive int value if the set object is greater
                than the specified object.
        """
